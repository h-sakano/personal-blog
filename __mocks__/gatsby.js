import React from 'react';

const gatsby = jest.requireActual('gatsby');
module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({
      activeClassName: _activeClassName,
      activeStyle: _activeStyle,
      getProps: _getProps,
      innerRef: _innerRef,
      partiallyActive: _partiallyActive,
      ref: _ref,
      replace: _replace,
      to,
      ...rest
    }) =>
      React.createElement('a', {
        ...rest,
        href: to,
      }),
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
