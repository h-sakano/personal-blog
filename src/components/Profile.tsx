import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';
import { Card, Space } from 'antd';
import { GithubOutlined, TwitterOutlined } from '@ant-design/icons';
import Img from 'gatsby-image';
import styles from './Profile.module.css';

const Profile = () => {
  const data = useStaticQuery(graphql`
    query {
      avatar: file(relativePath: { eq: "profile.png" }) {
        childImageSharp {
          fixed(width: 64) {
            ...GatsbyImageSharpFixed
          }
        }
      }
    }
  `);

  return (
    <Card className={styles.profile} title="プロフィール" bordered={false}>
      <div className={styles.avatarContainer}>
        <Img
          className={styles.avatar}
          fixed={data.avatar.childImageSharp.fixed}
        />
        <div className={styles.profileText}>
          <a
            className={styles.snsContainer}
            href="https://twitter.com/h_sakano"
            rel="noreferrer noopener"
            target="_blank"
          >
            <Space>
              <TwitterOutlined className={styles.twitter} />
              @h_sakano
            </Space>
          </a>
          <a
            className={styles.snsContainer}
            href="https://github.com/h-sakano"
            rel="noreferrer noopener"
            target="_blank"
          >
            <Space>
              <GithubOutlined className={styles.github} />
              @h-sakano
            </Space>
          </a>
        </div>
      </div>
    </Card>
  );
};

export default Profile;
