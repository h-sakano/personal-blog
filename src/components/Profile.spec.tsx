import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import * as React from 'react';
import Profile from './Profile';

beforeEach(() => {
  (useStaticQuery as jest.Mock<any, any>).mockImplementationOnce(() => ({
    avatar: {
      childImageSharp: {
        gatsbyImageData: {
          layout: 'fixed',
          backgroundColor: '#e8e8e8',
          images: {
            fallback: {
              src: '/static/7f05c1d57888914478fcf2c29a77ba96/1c9ce/profile.png',
              srcSet:
                '/static/7f05c1d57888914478fcf2c29a77ba96/1c9ce/profile.png 64w',
              sizes: '64px',
            },
            sources: [
              {
                srcSet:
                  '/static/7f05c1d57888914478fcf2c29a77ba96/8257c/profile.webp 64w',
                type: 'image/webp',
                sizes: '64px',
              },
            ],
          },
          width: 64,
          height: 64,
        },
      },
    },
    site: {
      siteMetadata: {
        title: 'サイトタイトル',
        description: 'サイト概要',
      },
    },
  }));
});

describe('<Layout />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Profile />);

    expect(asFragment()).toMatchSnapshot();
  });
});
