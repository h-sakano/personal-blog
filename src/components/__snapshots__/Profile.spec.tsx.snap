// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-card w-full"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          プロフィール
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="flex items-center"
      >
        <div
          class="block rounded-full gatsby-image-wrapper"
          style="position: relative; overflow: hidden; display: inline-block; width: 64px; height: 64px;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAAErUlEQVQ4yzXPe1SSdxgH8Pe/7ZxVJzs6p9bUpUddW7Us3VEH6VJzaSlhSXa6aTpRQanAnFOnzmwy07yntYla5qVIpdREJS9TAgQiLyCFmIrAe8Vrs9NerD3n+9/vfH7f5wEwSG+C8RhMsNEcxIhBhkUMXsbAGZVC1M/vaWs2zmtXVhZhSI/iryiMIBAeFIUBHG9kATN7g/kvxIhCupqCbGpogIebM+FrJ9qJYMlw/8qKCYYM6P8YwxAAhRZQUIeCCxteDxvn/11bLMlN97D/nEz08v7G9RjR23ePW9KF0whsQBDwg0QQGMNQAGeIGesQow4yzOFZwownAom7t1ud9PMJ8/U57PW9v6c7JSRwYU67aMJgGNzYGTFjWD8HmTOLX76+alpfM6HgLHHfLrutn7nYWXvucjnmRyR47Ccd/lEhFeEAhox47UcM6t4Y5qYRw5xC+vzXtNT4+PhTERR3ZwcHq63bLTbvsLRwcfjS+4C734Hde/fs5ff2vl1bgSC82TzAwptXc9OqRUj3sLF+05ZttnYOtjt2ujrahx1wOui2w85ii5utpb/nXrI/wcLS5lFb+/r6WxDEN8ebMWBeMzmrHkd0mu4n3J1OrjudXR0cnQ4dJuUnns2jEKgBHgUxJwWcUiYzdcs263sN99+9W/+Azc2z6pdalcIwoxrs7bCyttm336usrFb2wpiRfIUdeTAnjNjLvjoq6O8QG64V1/H7BlZXzWt/xDNK+fSkbEEz+by/m05nCYWvRaJ5fvfEpcvsqKP+rCDPexms8iZxTmVffadKpobhDflhAM346OsxiVYpm1TIpKPawSFta7u89ZGwqKiRkVoUn5CaklWVV93DzOZklXU/FKgw0yIEgnjt8vIyoFY8VytEmgnJmHyiuXkoM/v2A568+YGwitNz6eqNzMJmdt0wLbWUymAz87m8kXkdaIIgo16vVyqVgEo+opSNqF+KWu63/83potILblY9buHJy2r4fxQ15lZ2MnProhJz6aklydm1dY/lwkkQQZHcvOvB5FPAuGRwTDw49UJUUVabllYYFZMeS80srmyt4PTdqGi/Xt11kX7tAi2PkXk76kp5fddUv2weRuGYuIR9PgGAQih4MSIYl/zzuG2gqLgpPa8uMCCcFv9LQSk35bcq8umksONRjLQS1vUGUiSj8emkdMqgnX1DpSXbOLgA0qEePKNDPQMCWSt3mPNA4h8UGUgMSqBlkiPpAYHk0LAzcUk5fmE/exNC0ot4fKlO+VqdkMTYvM0GkAzwRc+eyoZ779Y0sgtbCurE7j+QCD4/kSh0L+8jvr5HIyITo2nZ3/kEu35LKGDfESumZSo1jXH5k083AcK+jpG+LvGzzluV93mcJuXTNnp03PHQM1Fn4rJYLAolNiIiPpaWHXo2OTEy/L1GpHs1ptDMxFAT7G2sgaHu9sHO1pHutuL8/BpqAFQdvfTo2pLgryV++fup3pUJYVleCYvd/Eo6CoqfzKoneK3cmrsNSZeZ56NjgD5eSw+3obOR089rukgh7//qi0O7HTvy6dPc0q47N7vu3vrz95xLKRnMpMTz586RjoeHhJKCjgSHhJ+kJSb8B2wr+bxurb9kAAAAAElFTkSuQmCC"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
          />
          <picture>
            <source
              srcset="/static/7f05c1d57888914478fcf2c29a77ba96/49ddd/profile.png 1x"
            />
            <img
              alt="プロフィール画像"
              height="64"
              loading="lazy"
              src="/static/7f05c1d57888914478fcf2c29a77ba96/49ddd/profile.png"
              srcset="/static/7f05c1d57888914478fcf2c29a77ba96/49ddd/profile.png 1x"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              width="64"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="/static/7f05c1d57888914478fcf2c29a77ba96/49ddd/profile.png 1x" /&gt;&lt;img loading="lazy" width="64" height="64" srcset="/static/7f05c1d57888914478fcf2c29a77ba96/49ddd/profile.png 1x" src="/static/7f05c1d57888914478fcf2c29a77ba96/49ddd/profile.png" alt="プロフィール画像" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <div
          class="flex-1 ml-3"
        >
          <a
            class="flex items-center"
            href="https://twitter.com/h_sakano"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <span
              aria-label="twitter"
              class="anticon anticon-twitter text-xl text-twitter"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="twitter"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
                />
              </svg>
            </span>
            <span
              class="ml-2 text-coloredLink"
            >
              @h_sakano
            </span>
          </a>
          <a
            class="flex items-center"
            href="https://github.com/h-sakano"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            <span
              aria-label="github"
              class="anticon anticon-github text-xl text-github"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="github"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                />
              </svg>
            </span>
            <span
              class="ml-2 text-coloredLink"
            >
              @h-sakano
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
